packages/core/logger/test/logger.spec.js

```diff
@@ -0,0 +1,57 @@
+import assert from 'assert';
+import sinon from 'sinon';
+import * as fs from 'fs';
+import { fakeEmbark } from 'embark-testing';
+import { Logger } from '../src/';
+import tmp from 'tmp';
+
+describe('core/logger', () => {
+
+  const { embark, plugins } = fakeEmbark();
+
+  let logger, testLogFn, logFile;
+
+  beforeEach(() => {
+    logFile = tmp.fileSync();
+
+    testLogFn = sinon.fake();
+    logger = new Logger({
+      events: embark.events,
+      logFunction: testLogFn,
+      logFile: logFile.name
+    });
+  });
+
+  afterEach(() => {
+    embark.teardown();
+    sinon.restore();
+  });
+
+  test('it should should use custom log function for logging', () => {
+    logger.info('Hello world');
+    assert(testLogFn.calledOnce);
+  });
+
+  test('it should inject color encoding based on log method', () => {
+    logger.info('Hello world');
+    assert(testLogFn.calledWith('\u001b[32mHello world\u001b[39m'));
+    logger.warn('Hello world');
+    assert(testLogFn.calledWith('\u001b[33mHello world\u001b[39m'));
+    logger.error('Hello world');
+    assert(testLogFn.calledWith('\u001b[31mHello world\u001b[39m'));
+  });
+
+  test('it should write logs to log file', () => {
+    logger.info('Some test log');
+    const logs = fs.readFileSync(logFile.name, 'utf8')
```

Comment:
I really think we should not be reading the file system here and instead faking `appendFileSync` (or whatever async equivalent we decide to go with). This will require `fs` to be configurable on `Logger` (obviously why we should be using DI, but I digress), so maybe the easiest option is to tack on an `fs` property in the constructor:
```
class Logger {
  constructor(options) {
    // ...
    this.fs = options.fs || fs;
    // ...
  }
}
```
And then in the test setup:
```
const appendFileSync = sinon.fake(); // could be async version depending on if you implement michael's suggestion
const fs = { appendFileSync };
logger = new Logger({
  events: embark.events,
  logFunction: testLogFn,
  logFile: "test"
  fs
});
```
and then in the test:
```
logger.info('Some test log');
sinon.assert.calledWith(appendFileSync, "test", ... rest of params);
```